lib/functoria/impl.ml

```diff
@@ -43,15 +43,9 @@ and 'a device = ('a, abstract) Device.t
 let abstract t = Abstract t
 
 let app_has_no_arguments = function
-  | App { args ; _ } ->
-    (match args with
-     | Cons _ -> false
-     | _ -> true)
-  | Dev { args ; _ } ->
-    (match args with
-     | Cons _ -> false
-     | _ -> true)
-  | If _ -> false
+  | App { args = Cons _ ; _ }
+  | Dev { args = Cons _ ; _ } -> false
+  | _ -> false
```

Comment:
Should be true :)

Reply:
true that is, I adjusted this in [b9b9ddb](https://github.com/mirage/mirage/pull/1428/commits/b9b9ddbb80b00d5762e82d8b62e339577d1667c0)